x-pack/platform/plugins/shared/inference/server/chat_complete/anonymization/anonymize_records.ts

```diff
@@ -46,13 +47,18 @@ export async function anonymizeRecords({
     (rule): rule is RegexAnonymizationRule => rule.type === 'RegExp'
   );
 
-  for (const rule of regexRules) {
-    state = await executeRegexRule({
-      rule,
-      state,
-      regexWorker,
-    });
-  }
+  const detectedRegexEntities = await executeRegexRules({
+    records: state.records,
+    rules: regexRules,
+    regexWorker,
+  });
+
+  // Process detected regex matches to resolve overlaps and apply masks
+  state = processMatches({
```

Comment:
No strong feeling about this one, but in my head we can probably rename this `processMatches` to something more descriptive... Maybe something like `maskAndAnonymizeMatches`? Ok to leave as is, just a thought.